deepfence_bootstrapper/supervisor/process.go

```diff
@@ -303,38 +327,40 @@ func WriteTo(dst, org string) error {
 	return err
 }
 
-func UpgradeProcessFromFile(name, path string) error {
+func UpgradeProcessFromFile(name, path string, restartSelf bool) error {
 	if name == SelfID {
 		return selfUpgradeFromFile(path)
 	}
 
 	access.RLock()
 	process, has := processes[name]
+	access.RUnlock()
 	if !has {
-		access.RUnlock()
 		return ErrPath
 	}
-	access.RUnlock()
 	process.access.Lock()
 	defer process.access.Unlock()
 
 	restart := false
 	if process.started {
-		log.Debug().Msg("Stop process")
+		log.Info().Msgf("Stop process started: %s", name)
 		err := process.stop()
 		if err != nil {
 			return err
 		}
+		log.Info().Msgf("Stop process done: %s", name)
 		restart = true
 	}
 
+	//process.path= /home/deepfence/bin/cloud_scanner
+	//path=/tmp/bins1655807479/cloud_scanner
 	err := WriteTo(process.path, path)
 	if err != nil {
 		return err
 	}
 
-	if restart {
-		log.Debug().Msg("Restart process")
+	if restart && !restartSelf {
```

Comment:
Yups self is outside the scope,  but this is to prevent the starting of the plugin processes (like cloudscanner, shipper etc) in case the bootstrapper itself needs to be upgraded. 
Reason being, If bootstrapper will be upgraded, it will start the plugin processes anyways when it restarts.